pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinestate.go

```diff
@@ -31,22 +31,7 @@ import (
 // or nil if no TaskRun should be created.
 func GetNextTask(prName string, state []*PipelineRunTaskRun, logger *zap.SugaredLogger) *PipelineRunTaskRun {
 	for _, prtr := range state {
-		if prtr.TaskRun != nil {
-			c := prtr.TaskRun.Status.GetCondition(duckv1alpha1.ConditionSucceeded)
-			if c == nil {
-				logger.Infof("TaskRun %s doesn't have a condition so it is just starting and we shouldn't start more for PipelineRun %s", prtr.TaskRunName, prName)
-				return nil
-			}
-			switch c.Status {
-			case corev1.ConditionFalse:
-				logger.Infof("TaskRun %s has failed; we don't need to run PipelineRun %s", prtr.TaskRunName, prName)
-				return nil
-			case corev1.ConditionUnknown:
-				logger.Infof("TaskRun %s is still running so we shouldn't start more for PipelineRun %s", prtr.TaskRunName, prName)
-				return nil
-			}
-		} else if canTaskRun(prtr.PipelineTask, state) {
-			logger.Infof("TaskRun %s should be started for PipelineRun %s", prtr.TaskRunName, prName)
+		if prtr.TaskRun == nil && canTaskRun(prtr.PipelineTask, state) {
```

Comment:
The unit test failure indicates that we are missing the cases where TaskRun is not nil, but hasn't run yet or failed

Reply:
will look in to this. I haven't looked at the test failures yet.

Reply:
i think the reason why this test was failing is because, i applied the follwing diff
```
@@ -61,7 +46,7 @@ func canTaskRun(pt *v1alpha1.PipelineTask, state []*PipelineRunTaskRun) bool {
                        for _, constrainingTaskName := range input.PassedConstraints {
                                for _, prtr := range state {
                                        // the constraining task must have a successful task run to allow this task to run
-                                       if prtr.Task.Name == constrainingTaskName {
+                                       if prtr.PipelineTask.Name == constrainingTaskName {
                                                if prtr.TaskRun == nil {
```
I changed the constrainingTaskName to match the `PipelineTask.Name` instead of `prtr.Task.Name`.
The reason being, you can have multiple tasks in your pipeline which point to same task name. The `PipelineTask.Name` is the unique identifier.

Reply:
i also changed this in the integration test for `helm-deploy`